examples/Makefile

```diff
@@ -1,25 +1,13 @@
+## this is a simple wrapper just to forward on known commands to the
+## embedding example with values pulled from Make.inc
 SRCDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
-BUILDDIR := .
 JULIAHOME := $(abspath $(SRCDIR)/..)
+BUILDDIR := .
 include $(JULIAHOME)/Make.inc
 
-outdir := $(libexecdir)
-
-embedding_binary := $(abspath $(outdir)/embedding$(JULIA_LIBSUFFIX)$(EXE))
-
-release: embedding
-debug: embedding-debug
-
-embedding: $(embedding_binary)
-embedding-debug: $(embedding_binary)
-
-$(embedding_binary): $(wildcard embedding/*)
-	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT)/examples/embedding $(JULIA_BUILD_MODE) \
-                                JULIA="$(bindir)/julia$(JULIA_LIBSUFFIX)$(EXE)" BIN="$(outdir)" \
-                                SPAWN="$(spawn)" CC="$(CC)"
-
-clean:
-	-rm -f $(embedding_binary) $(embedding_binary)-debug
-
-.PHONY: all embedding clean
-
+release: # default target
+# forward all variables expected by the embedding example
+JULIA:=$(call spawn,$(JULIA_EXECUTABLE))
+BIN:=$(build_bindir)
```

Comment:
> if you run in the right order, `make install` then `make -C examples` then run `test/embedding.jl` from inside the install tree (either via `Base.runtests("embedding")` or `test/runtests.jl embedding`), it works

That's not exactly common, but `make check` in the embedding makefile is even less so.

> I don't see the issue.

We want to prepare for a future setup where they will be installed. When that happens, it shouldn't be installed into bindir. It would make more sense to build it into build_libexecdir if we plan on eventually installing it into libexecdir. If your response to this sentence is going to be "that's broken," please explain exactly why.